### DISCUSSION TITLE
ModuleNotFoundError: No module named 'utils'

### DISCUSSION THREAD

**Discussion by meng25meng** [iiasa/message_ix#827] [MESSAGE modeling]
Hello, when I read in excel, I encountered difficulties in importing utilities from utils.py in the second step. Since I was using an older version, I needed to install util.py to run it. The installation file can no longer be found on the official website, and the reading method has changed. The data table has been prepared, if you change back to the current reading mode may need to change the table structure again, the workload is larger, if you want to download the new version of util, will it be compatible with the previous one? there is no ix_type_mapping to create table.
`import utils
ap = utils.add_par(scenario, horizon, vintage_years, firstyear, disp_error)
input_rows = utils.apply_filters(tecs, filters={'Parameter': 'input'})
ap.add_tec_input(input_rows)ModuleNotFoundError                       Traceback (most recent call last)
Cell In[23], line 1
----> 1 import utils
      2 ap = utils.add_par(scenario, horizon, vintage_years, firstyear, disp_error)`

**Comment by meng25meng** [comment]
It should be the version around 2019, but now you can't find this version, do you have any good solutions?

**Comment by glatterf42** [comment]
Thank you for providing additional information in #830.
